### ISSUE TITLE
HELP WANTED !!!

### ISSUE THREAD

**Issue by vedantmgoyal2009** [vedantmgoyal2009/test-gh-action#1]
@trenly if you have some spare time, please help me and tell me where the error is happening. 

this is a test repo I have set up for testing the GitHub action. the source code repo is at https://github.com/vedantmgoyal2009/winget-releaser/.

i'm making a api request using octokit and i don't know why but that part of the code doesn't get executed even.

i tried aync/await (https://github.com/vedantmgoyal2009/winget-releaser/commit/76b272ea65008f3bc9f8cba7b373c8c799edee77 and https://github.com/vedantmgoyal2009/winget-releaser/commit/67380cddb5202f3c3e7520dabcbe90c39bf031ae) but it doesn't seems to work.

**Comment by vedantmgoyal2009** [comment]
@itzlevvie if you can also give some guidance.

**Comment by ThomasThelen** [comment]
Is this about the GitHub action failing? 

From the error message [here](https://github.com/vedantmgoyal2009/test-gh-action/runs/4628252848?check_suite_focus=true) it has to `OhMyPosh` missing from the package. I see that it's defined in your github workflow file, but ` Error: ENOENT: no such file or directory, scandir 'winget-pkgs\manifests\j\JanDeDobbeleer\OhMyPosh'` makes me want to think that it's not in the right place for `winget-pkgs` (I'm not sure how windows packaging works).

To me, it looks like the node `fs` library is using it? (from the `node:internal/fs/utils:344` error message). I would define it in your package.json and remove it from the github workflow definition

**Comment by vedantmgoyal2009** [comment]
@ItzLevvie Hi! I'm using the following function in NodeJS Code to download files from GitHub Releases.
```js
async function downloadFile (downurl, filename) {
  https.get(downurl, (res) => {
    downfile = fs.createWriteStream(filename)
    res.pipe(downfile)
    downfile.on('finish', () => {
      downfile.close()
    })
  })
}
```
But I always get inside the file is a text response:

![image](https://user-images.githubusercontent.com/83997633/148537170-53ef47a2-7808-4d36-ae4a-79d2b0bead3c.png)

I prefer writing my own code instead of using third-party libraries like stephan83/download-github-release.

**Comment by ItzLevvie** [comment]
Node.js' `https.get` doesn't support URL redirects.

I was able to workaround it with this:
```js
const fs = require("fs");
const https = require("https");

async function downloadFile(URL, FileName) {
	let PermanentURL = URL;
	https.get(PermanentURL, (response) => {
		console.log("[DEBUG]: Response Code: " + response.statusCode);
		if (response.statusCode === 400 || response.statusCode === 403 || response.statusCode === 404) {
			return;
		} else if (response.statusCode === 301 || response.statusCode === 302) {
			let TemporaryURL = response.headers["location"];
			https.get(TemporaryURL, (response) => {
				console.log("[DEBUG]: URL (with redirects): " + PermanentURL);
				console.log("[DEBUG]: URL (without redirects): " + TemporaryURL);
				console.log("[DEBUG]: File: " + FileName);
				console.log();
				console.log("[DEBUG]: Downloading " + FileName);
				downloadFile = fs.createWriteStream(FileName);
				response.pipe(downloadFile);
				downloadFile.on("finish", () => {
					console.log("[DEBUG]: Downloaded " + FileName);
					downloadFile.close();
				});
			});
		} else {
			https.get(PermanentURL, (response) => {
				console.log("[DEBUG]: URL (without redirects): " + PermanentURL);
				console.log("[DEBUG]: File: " + FileName);
				console.log();
				console.log("[DEBUG]: Downloading " + FileName);
				downloadFile = fs.createWriteStream(FileName);
				response.pipe(downloadFile);
				downloadFile.on("finish", () => {
					console.log("[DEBUG]: Downloaded " + FileName);
					downloadFile.close();
				});
			});
		}
	});
}

downloadFile("https://github.com/git-for-windows/git/releases/download/v2.34.1.windows.1/Git-2.34.1-64-bit.exe", "Git-2.34.1-64-bit.exe");
```

but you're probably better off using another module for https (since the code I wrote is likely bad for performance reasons)

**Comment by jedieaston** [comment]
If you want a less hacky way to do the same thing, the `follow-redirects` module will do it. `npm install follow-redirects`, then change `const https = require("https")` to `const https = require('follow-redirects').https`. But I see no issues with Levvie's code, at least without running it.

**Comment by vedantmgoyal2009** [comment]
Yeah! I think I can use that. I use dependencies as little as I can because earlier, I used third-party modules many times, then got to know that they were no longer maintained and had vulnerabilities, and finally I removed them and then wrote my own code after removing them.

**Comment by jedieaston** [comment]
That's usually a good attitude to have, but somewhat tough in the Node.js ecosystem since everyone assumes you are downloading tons of stuff from npm all the time. In this case I think it's probably safe.

**Comment by vedantmgoyal2009** [comment]
Yes, it also seems that the developer is actively maintaining the module👍
